File: doc/td/demo/src/main/java/com/ceremonie/demo/services/interfaces/ContributionService.java
package com.ceremonie.demo.services.interfaces;


import java.util.List;

import com.ceremonie.demo.dto.request.CreateContributionRequest;
import com.ceremonie.demo.dto.response.ContributionResponse;

public interface ContributionService {
    ContributionResponse createContribution(CreateContributionRequest request);
    ContributionResponse updateContribution(Long id, CreateContributionRequest request);
    ContributionResponse getContributionById(Long id);
    List<ContributionResponse> getAllContributions();
    List<ContributionResponse> getContributionsByMember(Long memberId);
    List<ContributionResponse> getContributionsByYear(Long yearId);
    List<ContributionResponse> getUnpaidContributions();
    void markAsPaid(Long contributionId, Long transactionId);
    void deleteContribution(Long id);
}

File: doc/td/ceremony-tresorier/src/services/contributionService.js
// src/services/contributionService.js
import api from './api';

const API_URL = import.meta.env.VITE_API_URL;

export const contributionService = {
  getContributions: () => api.get(`${API_URL}/contributions`),
  
  getContributionsByMember: (memberId) => 
    api.get(`${API_URL}/contributions/member/${memberId}`),
  
  getUnpaidContributions: () => 
    api.get(`${API_URL}/contributions/unpaid`),
  
  createContribution: (contributionData) => 
    api.post(`${API_URL}/contributions`, contributionData),
  
  updateContribution: (id, contributionData) => 
    api.put(`${API_URL}/contributions/${id}`, contributionData),
  
  markAsPaid: (contributionId, transactionId) => 
    api.put(`${API_URL}/contributions/${contributionId}/mark-paid/${transactionId}`, {}),
};